src/devices/Common/Interop/Unix/Libc/Interop.libc.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Disable these StyleCop rules for this file, as we are using native names here.
+#pragma warning disable SA1300 // Element should begin with upper-case letter
+#pragma warning disable CS1591 // Public member is not documented - These members are not public in the final package
+
+using System;
+using System.Runtime.InteropServices;
+
+#if BUILDING_IOT_DEVICE_BINDINGS
+internal
+#else
+public
+#endif
+partial class Interop
+{
+    [DllImport(LibcLibrary, SetLastError = true)]
+    public static extern int ioctl(int fd, int request, IntPtr argp);
+
+    [DllImport(LibcLibrary, SetLastError = true)]
+    public static extern int open([MarshalAs(UnmanagedType.LPStr)] string pathname, FileOpenFlags flags);
+
+    [DllImport(LibcLibrary)]
+    public static extern int close(int fd);
+
+    [DllImport(LibcLibrary, SetLastError = true)]
+    public static extern IntPtr mmap(IntPtr addr, int length, MemoryMappedProtections prot, MemoryMappedFlags flags, int fd, int offset);
+
+    [DllImport(LibcLibrary)]
+    public static extern int munmap(IntPtr addr, int length);
+}
+
+#if BUILDING_IOT_DEVICE_BINDINGS
+internal
+#else
+public
+#endif
+    enum FileOpenFlags
+{
+    O_RDONLY = 0x00,
+    O_RDWR = 0x02,
+    O_NONBLOCK = 0x800,
+    O_SYNC = 0x101000
+}
+
+[Flags]
+#if BUILDING_IOT_DEVICE_BINDINGS
+internal
+#else
+public
+#endif
+    enum MemoryMappedProtections
+{
+    PROT_NONE = 0x0,
+    PROT_READ = 0x1,
+    PROT_WRITE = 0x2,
+    PROT_EXEC = 0x4
+}
+
+[Flags]
+
+#if BUILDING_IOT_DEVICE_BINDINGS
+internal
+#else
+public
+#endif
+    enum MemoryMappedFlags
```

Comment:
perhaps we could make all of these subclasses nested so that we don't have as many ifdefs

Reply:
Done